packages/grafana-prometheus/src/components/usePromQueryFieldEffects.ts

```diff
@@ -0,0 +1,64 @@
+import { MutableRefObject, useEffect, useRef } from 'react';
+
+import { DataFrame, DateTime, TimeRange } from '@grafana/data';
+
+import PromQlLanguageProvider from '../language_provider';
+import { roundMsToMin } from '../language_utils';
+
+import { CancelablePromise } from './cancelable-promise';
+
+export function usePromQueryFieldEffects(
+  languageProvider: PromQlLanguageProvider,
+  range: TimeRange | undefined,
+  series: DataFrame[] | undefined,
+  refreshMetrics: (languageProviderInitRef: MutableRefObject<CancelablePromise<unknown> | null>) => Promise<void>,
+  refreshHint: () => void
+) {
+  const lastRangeRef = useRef<{ from: DateTime; to: DateTime } | null>(null);
+  const languageProviderInitRef = useRef<CancelablePromise<unknown> | null>(null);
+
+  // Effect for initial load
+  useEffect(() => {
+    if (languageProvider) {
```

Comment:
Isn't this always true as `languageProvider: PromQlLanguageProvider` and there can't be undefined/null?

Reply:
That is coming from the legacy code https://github.com/grafana/grafana/pull/101515/files#diff-6988fc0e164953875df06146908627a06100c249f4e46c37a743d85279161baaL67-L69
But you are right. we don't need that.

Reply:
https://github.com/grafana/grafana/pull/101515/commits/1760e8d1a9fb526de218732414e917fd0666258f